src/commands/server/restore.ts

```diff
@@ -0,0 +1,126 @@
+import { requestRestore } from '../../api/backup-restore'
+import { cli } from 'cli-ux'
+import { ApiTasks } from '../../tasks/platforms/api'
+import { getCommandSuccessMessage, notifyCommandCompletedSuccessfully, wrapCommandError } from '../../util'
+import { V1CheClusterRestore, V1CheClusterRestoreStatus } from '../../api/typings/backup-restore-crds'
+
+import { awsAccessKeyId, awsSecretAccessKey, AWS_ACCESS_KEY_ID_KEY, AWS_SECRET_ACCESS_KEY_KEY, backupRepositoryPassword, backupRepositoryUrl, backupRestServerPassword, backupRestServerUsername, backupServerConfigName, BACKUP_REPOSITORY_PASSWORD_KEY, BACKUP_REPOSITORY_URL_KEY, BACKUP_REST_SERVER_PASSWORD_KEY, BACKUP_REST_SERVER_USERNAME_KEY, BACKUP_SERVER_CONFIG_CR_NAME_KEY, getBackupServerConfiguration, sshKey, sshKeyFile, SSH_KEY_FILE_KEY, SSH_KEY_KEY } from './backup'
+
+export default class Restore extends Command {
+  static description = 'Restore Eclipse Che installation'
+
+  static examples = [
+    '# Reuse existing backup configuration:\n' +
+    'chectl server:restore',
+    '# Restore from specific backup snapshot using previos backup configuration:\n' +
+    'chectl server:restore -s 585421f3',
+    '# Create and use configuration for REST backup server:\n' +
+    'chectl server:resotre -r rest:http://my-sert-server.net:4000/che-backup -p repopassword',
+    '# Create and use configuration for AWS S3 (and API compatible) backup server (bucket should be precreated):\n' +
+    'chectl server:backup -r s3:s3.amazonaws.com/bucketche -p repopassword',
+    '# Create and use configuration for SFTP backup server:\n' +
+    'chectl server:backup -r=sftp:user@my-server.net:/srv/sftp/che-data -p repopassword',
+  ]
+
+  static flags: flags.Input<any> = {
+    help: flags.help({ char: 'h' }),
+    chenamespace: cheNamespace,
+    [BACKUP_REPOSITORY_URL_KEY]: backupRepositoryUrl,
+    [BACKUP_REPOSITORY_PASSWORD_KEY]: backupRepositoryPassword,
+    [BACKUP_REST_SERVER_USERNAME_KEY]: backupRestServerUsername,
+    [BACKUP_REST_SERVER_PASSWORD_KEY]: backupRestServerPassword,
+    [SSH_KEY_KEY]: sshKey,
+    [SSH_KEY_FILE_KEY]: sshKeyFile,
+    [AWS_ACCESS_KEY_ID_KEY]: awsAccessKeyId,
+    [AWS_SECRET_ACCESS_KEY_KEY]: awsSecretAccessKey,
+    [BACKUP_SERVER_CONFIG_CR_NAME_KEY]: backupServerConfigName,
+    'snapshot-id': string({
+      char: 's',
+      description: 'ID of a snapshot to restore from',
+      required: false,
+    }),
+  }
+
+  async run() {
+    const { flags } = this.parse(Restore)
+    flags.chenamespace = flags.chenamespace || DEFAULT_CHE_NAMESPACE
+    const ctx = await ChectlContext.initAndGet(flags, this)
+
+    await this.config.runHook(DEFAULT_ANALYTIC_HOOK_NAME, { command: Restore.id, flags })
+
+    const tasks = new Listr([], ctx.listrOptions)
+    const apiTasks = new ApiTasks()
+    tasks.add(apiTasks.testApiTasks(flags))
+    tasks.add(this.getRestoreTasks(flags))
+    // Export self-signed CA cert if any
+    flags.tls = true
```

Comment:
Not sure if I understand this.

Reply:
`retrieveCheCaCertificateTask` has `enabled: () => flags.tls && flags.installer !== 'helm'`
That's why I added `tls` flag.

Reply:
pls remove retrieving certificate.
there is a dedicated command cacert:export

Reply:
ok